### PR TITLE
Update password backend

### PR DESCRIPTION
**Pull request by DmytroDovhal**

### CODE REVIEW

File: src/main/java/greencity/config/SecurityConfig.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 "/googleSecurity/**",
                 "/place/filter/**",
                 "/restorePassword/**",
-                "/updatePassword/**"
+                "/changePassword/**"
             ).permitAll()
```
Comment (GuN41K) [inline src/main/java/greencity/config/SecurityConfig.java]: Does unregistered user can change password? Which password?

File: src/main/java/greencity/exception/PasswordsDoNotMatchesException.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package greencity.exception;
+
+/**
+ * Exception that we get when user trying to verify email with bad token.
+ *
```
Comment (GuN41K) [inline src/main/java/greencity/exception/PasswordsDoNotMatchesException.java]: Fix java docs.

File: src/main/java/greencity/config/SecurityConfig.java (hunk 1 above)
Comment (DmytroDovhal) [reply src/main/java/greencity/config/SecurityConfig.java]: It's for forgot password logic, when your restore your password are not logined

File: src/main/java/greencity/exception/PasswordsDoNotMatchesException.java (hunk 2 above)
Comment (DmytroDovhal) [reply src/main/java/greencity/exception/PasswordsDoNotMatchesException.java]: Ok)

File: src/main/java/greencity/security/dto/ownsecurity/OwnUpdateDto.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class OwnUpdateDto {
```
Comment (GuN41K) [inline src/main/java/greencity/security/dto/ownsecurity/OwnUpdateDto.java]: I thing that it is not good name for change password DTO

File: src/main/java/greencity/security/repository/OwnSecurityRepo.java [hunk 4]
```diff
@@ -25,4 +26,12 @@
...
     @Query("update OwnSecurity o set o.password = :password where o.user.id = :id")
     void updatePassword(@Param("password") String password, @Param("id") Long id);
+
+    /**
+     * Method for finding user password.
+     *
+     * @param user - {@link User}
+     * @return
+     */
+    OwnSecurity findByUser(User user);
```
Comment (GuN41K) [inline src/main/java/greencity/security/repository/OwnSecurityRepo.java]: If you have already user entity in service, user has own security! You can get it by using getter for ownSecurity

File: src/main/java/greencity/security/repository/RestorePasswordEmailRepo.java [hunk 5]
```diff
@@ -12,6 +13,16 @@
...
     Optional<RestorePasswordEmail> findByToken(String token);
+
+    /**
+     * Method for finding by user.
+     *
+     * @param user - {@link User} - user for search
+     * @return - {@link RestorePasswordEmail}
+     * @author Dmytro Dovhal
+     */
+    RestorePasswordEmail findByUser(User user);
```
Comment (GuN41K) [inline src/main/java/greencity/security/repository/RestorePasswordEmailRepo.java]: Same like with OwnSecurity

File: src/main/java/greencity/security/repository/OwnSecurityRepo.java (hunk 4 above)
Comment (DmytroDovhal) [reply src/main/java/greencity/security/repository/OwnSecurityRepo.java]: Ok, nice comment, really! i'l fix it.

File: src/main/java/greencity/security/repository/RestorePasswordEmailRepo.java (hunk 5 above)
Comment (DmytroDovhal) [reply src/main/java/greencity/security/repository/RestorePasswordEmailRepo.java]: fixed

**Pull request merged**